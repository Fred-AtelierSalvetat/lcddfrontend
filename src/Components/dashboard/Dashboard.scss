$navbarwidth: 80px;

#lcdd-dashboard {
    display: flex;

    height: calc(100vh - 72px);

    .navbar {
        overflow-y: auto;
        overflow-x: hidden;
        align-items: flex-start !important;
        min-width: $navbarwidth;
        width: $navbarwidth;
        height: 100%;
        margin: 0;
        padding: 46px 1px 0 1px;
        background-color: var(--admin-nav-dark);

        .navbar-nav {
            min-width: calc(#{$navbarwidth} - 2px);
            max-width: calc(#{$navbarwidth} - 2px);
            width: calc(#{$navbarwidth} - 2px);
            display: flex;
            justify-items: center;
            .nav-link {
                pointer-events: auto;
                margin: 10px 0;
                padding: 0;
                background: var(--admin-nav-dark);
                &.disabled {
                    pointer-events: none;
                    .button-panel {
                        p {
                            color: var(--gray);
                        }
                        svg {
                            fill: var(--gray);
                        }
                    }
                }
                &.active {
                    pointer-events: none;
                    background: var(--white);
                    border-radius: 2px;
                    .button-panel {
                        p {
                            color: var(--admin-nav-dark);
                        }
                        svg {
                            fill: var(--admin-nav-dark);
                        }
                    }
                }
                .button-panel {
                    transition: transform 0.2s;

                    &:hover {
                        transform: scale(1.1);
                    }

                    p {
                        margin: 8px 0 0 0;
                        text-align: center;
                        color: var(--white);
                        font-weight: 400;
                        font-size: small;
                    }
                    svg {
                        width: 30px;
                        height: 30px;
                        fill: var(--white);
                    }
                }
            }
        }
    }

    #lcdd-dashboard-page {
        height: 100%;
        width: calc(100% - #{$navbarwidth});
        padding: 0 36px;
        background: linear-gradient(180deg, #ffffff 0%, #ffffff 23.96%, #f2f2f2 100%);
        overflow: auto;
    }
}
