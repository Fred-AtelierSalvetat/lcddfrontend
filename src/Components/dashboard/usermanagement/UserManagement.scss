$line-separator-color: #eceeef;
$headers-color: #eceeef;
$header-background-color: #f2f2f2;
$input-border-color: #cccccc;
$input-place-holder-color: #999999;
$awaiting-validation-background-color: #f0ad4e;

#lcdd-dashboard-page {
    .navPill {
        &-enter {
            background-color: yellowgreen !important;
        }

        &-enter-active {
            background-color: blue !important;
            transition: background-color 2000ms !important;
        }
        &-exit {
            opacity: 1;
            background-color: yellow !important;
            transition: opacity 1000ms;
        }

        &-exit-active {
            opacity: 1;
            background-color: yellow !important;
            transition: opacity 1000ms;
        }
    }
    .table_and_filters_pane {
        background-color: $header-background-color;
        border-top-left-radius: 0.25rem;
        border-top-right-radius: 0.25rem;
        //width: fit-content;
        min-width: 100%;
        filter: drop-shadow(0 0.25rem 0.5rem rgba(130, 130, 130, 0.5));
        .nav {
            padding: 0.5rem;

            .nav-link {
                transition: transform 0.2s;
                padding: 0.5rem;
                color: var(--primary);
                background-color: transparent;
                &:hover {
                    transform: scale(1.1);
                }
                &.nav-link.active {
                    color: var(--white);
                    background-color: var(--primary);
                    &:hover {
                        transform: none;
                    }
                }
            }
            .searchbox {
                width: calc(100vw - 160px);
                position: relative;
                align-self: bottom;
                //TODO
                @media (min-width: 769px) {
                    margin-left: auto;
                    min-width: 200px;
                    width: 400px;
                    align-self: top;
                }
                input {
                    padding: 0.5rem 1rem;
                    border: 0;
                    font-size: 1rem;
                    font-style: normal;
                    font-weight: 400;
                    line-height: 1.5rem;
                    letter-spacing: 0;
                    text-align: left;
                    border: 1px solid $input-border-color;
                    &::placeholder {
                        color: $input-place-holder-color;
                    }
                }

                svg {
                    position: absolute;
                    right: 10px;
                    top: 13px;
                    z-index: 9999;
                }
            }
        }

        .tab-content {
            background-color: var(--white);
            margin-bottom: 7px;
            .tab-pane {
                height: calc(100vh - 200px);
                min-height: 97px;
                overflow: auto;
            }
            thead {
                tr {
                    th {
                        position: sticky;
                        top: 0;
                        z-index: 10;
                        background-color: var(--gray-dark);
                        color: $headers-color;
                        padding: 0.75rem 1rem;
                        font-style: normal;
                        font-weight: 700;
                        line-height: 1, 5rem;
                        letter-spacing: 0;
                        text-align: left;
                        vertical-align: middle;
                        white-space: nowrap;
                        > .CenteredHeader {
                            text-align: center;
                        }
                    }
                }
            }
            tbody {
                tr {
                    border-top: 1px solid $line-separator-color;
                    td {
                        z-index: 9;
                        padding: 0.75rem 1rem;
                        background-color: var(--white);
                        .Centered {
                            display: flex;
                            align-items: center;
                            justify-content: center;

                            .speakerStatus {
                                color: var(--white);
                                border-radius: 20px;

                                height: 20px;
                                width: 68px;
                                display: flex;
                                align-items: center;
                                justify-content: center;
                                margin: 4px 0 6px 0;
                                &.speakerAwaitingAnswer {
                                    background-color: var(--cyan);
                                }
                                &.speakerAwaitingValidation {
                                    background-color: $awaiting-validation-background-color;
                                }
                                p {
                                    margin: 0;
                                    line-height: 22px;
                                    font-size: 12px;
                                    font-style: normal;
                                    font-weight: 700;
                                    letter-spacing: 0px;
                                    white-space: nowrap;
                                }
                            }
                        }

                        > p {
                            margin: 0;
                            font-style: normal;
                            font-weight: 400;
                            line-height: 1.5rem;
                            letter-spacing: 0px;
                            text-align: left;
                            vertical-align: middle;
                            white-space: nowrap;
                        }
                        > a {
                            font-size: small;
                            font-weight: 400;
                            font-style: normal;
                            white-space: nowrap;
                            color: var(--link);
                        }
                        .ActionMenuIcon {
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            cursor: pointer;
                        }
                        .userStatus {
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            > svg {
                                cursor: pointer;

                                &:hover {
                                    transform: scale(1.1);
                                }
                                #toggle_on_24px {
                                    fill: var(--success);
                                }
                                #toggle_off_24px {
                                    fill: var(--danger);
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

.arrow {
    display: none !important;
}

.popover-body {
    margin: 0;
    padding: 0 !important;
    border-width: 1px;
    display: flex;
    flex-direction: column;
    > .actionButton {
        display: flex;
        align-items: center;
        padding: 0.5rem 1rem;
        border-bottom: 1px solid var(--gray-light);
        cursor: pointer;

        &:nth-last-child(1) {
            border-bottom: none;
        }

        .actionButtonPanel {
            transition: transform 0.2s;
            &:hover {
                transform: scale(1.1);
            }

            span {
                height: 1.5rem;
                margin: 0;
                padding-left: 0.5rem;
                font-weight: 400;
                font-style: normal;
            }
        }
    }
}
