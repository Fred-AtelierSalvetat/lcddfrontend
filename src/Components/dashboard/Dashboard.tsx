import React, { Fragment, FC, useState, useEffect } from 'react';
import { useParams, useHistory, Redirect } from 'react-router-dom';
import { ReactComponent as WorkShopsIcon } from '~/assets/icons/workshops_24px.svg';
import { ReactComponent as WorkshopCreationIcon } from '~/assets/icons/add_24px.svg';
//import { ReactComponent as QuestionsIcon } from '~/assets/icons/questions_24px.svg';
import { ReactComponent as UsersIcon } from '~/assets/icons/users_24px.svg';
//import { ReactComponent as AnalyticsIcon } from '~/assets/icons/analytics_24px.svg';
// import { ReactComponent as NotificationsIcon } from '~/assets/icons/notifications_24px.svg';
// import { ReactComponent as HelpIcon } from '~/assets/icons/help_24px.svg';
import ErrorBoundary from '~/Components/shared/ErrorBoundary';
import UserManagement from './usermanagement/UserManagement';
import NewWorkshop from './newWorkshop/NewWorkshop';
import Workshops from './workshops/Workshops';
import EditWorkshop from './editWorkshop/EditWorkshop';
import AlertNotificationBox from './AlertNotificationBox';
import './Dashboard.scss';

const Dashboard: FC = () => {
    const [selectedPageParam, setSelectedPageParam] = useState('');
    const { selectedPage } = useParams() as {
        selectedPage: string;
    };
    useEffect(() => setSelectedPageParam(selectedPage), [selectedPage]);

    const history = useHistory();

    interface PageDesc {
        icon: JSX.Element;
        label: JSX.Element;
        href: string;
        page: JSX.Element;
    }
    const pages: PageDesc[] = [
        {
            icon: <WorkShopsIcon />,
            label: <p>Ateliers</p>,
            href: '/dashboard/workshops',
            page: <Workshops />,
        },
        // {
        //     icon: <QuestionsIcon />,
        //     label: <p>Questions</p>,
        //     href: '#/dashboard/questions',
        //     page: <p>Questions</p>,
        // },
        {
            icon: <WorkshopCreationIcon />,
            label: (
                <p>
                    Cr√©er
                    <br />
                    un atelier
                </p>
            ),
            href: '/dashboard/newWorkshop',
            page: <NewWorkshop />,
        },
        {
            icon: <UsersIcon />,
            label: (
                <p>
                    Gestion
                    <br />
                    utilisateurs
                </p>
            ),
            href: '/dashboard/users',
            page: <UserManagement />,
        },
        {
            href: '/dashboard/editWorkshop',
            page: <EditWorkshop />,
        },
        // {
        //     icon: <AnalyticsIcon />,
        //     label: <p>Analytics</p>,
        //     href: '#/dashboard/analytics',
        //     page: <p>Analytics</p>,
        //     disabled: true,
        // },
        // {
        //     icon: <NotificationsIcon />,
        //     label:  <p>Notifications</p>,
        //     href: '#/dashboard/notifications',
        //     page: <p>Notifications</p>,
        //     disabled: true,
        // },
        // {
        //     icon: <HelpIcon />,
        //     label: <p>Aide</p>,
        //     href: '#/dashboard/help',
        //     page: <p>Help</p>,
        //     disabled: true,
        // },
    ];

    const isSelectedPage = (page_desc: PageDesc) => page_desc.href.split('/').slice(-1)[0] === selectedPageParam;

    const get_active_page = () => {
        const active_page_desc = pages.find((page_desc) => isSelectedPage(page_desc));
        return active_page_desc ? active_page_desc.page : null;
    };

    const activePage: JSX.Element | null = get_active_page();

    if (selectedPageParam && !activePage) {
        return <Redirect to="/no-match" />;
    }

    return (
        <ErrorBoundary>
            <div id="lcdd-dashboard">
                <div id="lcdd-dashboard-navbar">
                    {pages
                        .filter((page) => !!page.icon)
                        .map((page_desc) => (
                            <Fragment key={page_desc.href}>
                                <input
                                    type="radio"
                                    id={page_desc.href}
                                    value={page_desc.href}
                                    checked={isSelectedPage(page_desc)}
                                    onChange={(changeEvent) => {
                                        history.push(changeEvent.target.value);
                                        setSelectedPageParam(page_desc.href.split('/').slice(-1)[0]);
                                    }}
                                />
                                <label htmlFor={page_desc.href}>
                                    <div>
                                        {page_desc.icon}
                                        {page_desc.label}
                                    </div>
                                </label>
                            </Fragment>
                        ))}
                    <div className="selection-sliding-background"></div>
                </div>
                <div id="lcdd-dashboard-page">
                    <AlertNotificationBox />
                    <div id="activePage">{activePage}</div>
                </div>
            </div>
        </ErrorBoundary>
    );
};

export default Dashboard;
