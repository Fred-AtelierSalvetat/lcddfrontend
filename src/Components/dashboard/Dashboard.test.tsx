import React from 'react';
import { expect } from 'chai';
import { mount } from 'enzyme';

import { MemoryRouter, Route } from 'react-router-dom';
import { NavLink } from 'react-bootstrap';

import Dashboard from './Dashboard';
import UserManagement from './usermanagement/UserManagement';

jest.mock('./usermanagement/UserManagement');

describe('<Dashboard />', () => {
    it('contains 6 navigation links', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/dashboard']}>
                <Route path="/dashboard" exact={true}>
                    <Dashboard />
                </Route>
            </MemoryRouter>,
        );
        expect(wrapper.find(NavLink)).to.have.length(6);
    });
    it('contains a link to #/dashboard/workshops', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/dashboard']}>
                <Route path="/dashboard" exact={true}>
                    <Dashboard />
                </Route>
            </MemoryRouter>,
        );
        expect(wrapper.find({ href: '#/dashboard/workshops' })).to.exist;
    });

    it('contains a link to #/dashboard/users', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/dashboard']}>
                <Route path="/dashboard" exact={true}>
                    <Dashboard />
                </Route>
            </MemoryRouter>,
        );
        expect(wrapper.find({ href: '#/dashboard/users' })).to.exist;
    });

    it('contains a link to #/dashboard/analytics', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/dashboard']}>
                <Route path="/dashboard" exact={true}>
                    <Dashboard />
                </Route>
            </MemoryRouter>,
        );
        expect(wrapper.find({ href: '#/dashboard/analytics' })).to.exist;
    });

    it('contains a link to #/dashboard/notifications', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/dashboard']}>
                <Route path="/dashboard" exact={true}>
                    <Dashboard />
                </Route>
            </MemoryRouter>,
        );
        expect(wrapper.find({ href: '#/dashboard/notifications' })).to.exist;
    });

    it('contains a link to #/dashboard/help', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/dashboard']}>
                <Route path="/dashboard" exact={true}>
                    <Dashboard />
                </Route>
            </MemoryRouter>,
        );
        expect(wrapper.find({ href: '#/dashboard/help' })).to.exist;
    });

    it("shall handle route's params to activate the corresponding NavLink", () => {
        const userManagementMockContent = <div>UserManagementTESTPAGE</div>;
        UserManagement.mockImplementation(() => userManagementMockContent);
        const wrapper = mount(
            <MemoryRouter initialEntries={['/dashboard/users']}>
                <Route path="/dashboard/:selectedPage" exact={true}>
                    <Dashboard />
                </Route>
            </MemoryRouter>,
        );
        expect(wrapper.find(NavLink).filter({ href: '#/dashboard/users' })).to.have.length(1);
        expect(wrapper.find(NavLink).filter({ href: '#/dashboard/users' }).at(0).props().active).to.equal(true);
        expect(wrapper.find('#lcdd-dashboard-page').childAt(0)).to.contain(userManagementMockContent);
    });

    it('should match its reference snapshot', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[{ pathname: '/dashboard', key: 'testKey' }]}>
                <Route path="/dashboard" exact={true}>
                    <Dashboard />
                </Route>
            </MemoryRouter>,
        );

        expect(wrapper).to.matchSnapshot();
    });
});
