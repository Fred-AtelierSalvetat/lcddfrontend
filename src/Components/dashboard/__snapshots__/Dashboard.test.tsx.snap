// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should match its reference snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/dashboard",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/dashboard",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/dashboard",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Route
      exact={true}
      path="/dashboard"
    >
      <Dashboard>
        <ErrorBoundary>
          <div
            id="lcdd-dashboard"
          >
            <Navbar
              collapseOnSelect={false}
              expand={true}
              variant="light"
            >
              <nav
                className="navbar navbar-expand navbar-light"
              >
                <Nav
                  className="flex-column"
                  fill={false}
                  justify={false}
                >
                  <ForwardRef
                    as="div"
                    className="flex-column navbar-nav"
                    onSelect={[Function]}
                  >
                    <div
                      className="flex-column navbar-nav"
                      onKeyDown={[Function]}
                    >
                      <NavLink
                        active={false}
                        as={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "displayName": "SafeAnchor",
                            "render": [Function],
                          }
                        }
                        disabled={false}
                        href="#/dashboard/newWorkshop"
                        key="#/dashboard/newWorkshop"
                      >
                        <ForwardRef
                          active={false}
                          as={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "displayName": "SafeAnchor",
                              "render": [Function],
                            }
                          }
                          className="nav-link"
                          disabled={false}
                          href="#/dashboard/newWorkshop"
                        >
                          <SafeAnchor
                            className="nav-link"
                            data-rb-event-key="#/dashboard/newWorkshop"
                            disabled={false}
                            href="#/dashboard/newWorkshop"
                            onClick={[Function]}
                          >
                            <a
                              className="nav-link"
                              data-rb-event-key="#/dashboard/newWorkshop"
                              href="#/dashboard/newWorkshop"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                            >
                              <div
                                className="button-panel"
                              >
                                <ForwardRef(SvgAdd24Px)>
                                  <svg>
                                    add_24px.svg
                                  </svg>
                                </ForwardRef(SvgAdd24Px)>
                                <p>
                                  Cr√©er
                                  <br />
                                  un atelier
                                </p>
                              </div>
                            </a>
                          </SafeAnchor>
                        </ForwardRef>
                      </NavLink>
                      <NavLink
                        active={false}
                        as={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "displayName": "SafeAnchor",
                            "render": [Function],
                          }
                        }
                        disabled={false}
                        href="#/dashboard/users"
                        key="#/dashboard/users"
                      >
                        <ForwardRef
                          active={false}
                          as={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "displayName": "SafeAnchor",
                              "render": [Function],
                            }
                          }
                          className="nav-link"
                          disabled={false}
                          href="#/dashboard/users"
                        >
                          <SafeAnchor
                            className="nav-link"
                            data-rb-event-key="#/dashboard/users"
                            disabled={false}
                            href="#/dashboard/users"
                            onClick={[Function]}
                          >
                            <a
                              className="nav-link"
                              data-rb-event-key="#/dashboard/users"
                              href="#/dashboard/users"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                            >
                              <div
                                className="button-panel"
                              >
                                <ForwardRef(SvgUsers24Px)>
                                  <svg>
                                    users_24px.svg
                                  </svg>
                                </ForwardRef(SvgUsers24Px)>
                                <p>
                                  Gestion
                                  <br />
                                  utilisateurs
                                </p>
                              </div>
                            </a>
                          </SafeAnchor>
                        </ForwardRef>
                      </NavLink>
                    </div>
                  </ForwardRef>
                </Nav>
              </nav>
            </Navbar>
            <div
              id="lcdd-dashboard-page"
            />
          </div>
        </ErrorBoundary>
      </Dashboard>
    </Route>
  </Router>
</MemoryRouter>
`;
